rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt

```diff
@@ -298,11 +299,21 @@ open class InternalInterpreter(
         // probably it is an error during the ast processing.
         // as workaround, if null assumes the number of lines in the compile compileTimeArray
         // as value for compileTimeRecordsPerLine
-        val lines = if (arrayType.compileTimeRecordsPerLine == null) compileTimeArray.lines.size else arrayType.compileTimeRecordsPerLine
+        val lines = arrayType.compileTimeRecordsPerLine ?: compileTimeArray.lines.size
         val l: MutableList<Value> =
             compileTimeArray.lines.chunkAs(lines, arrayType.element.size)
                 .map {
-                    coerce(StringValue(it), arrayType.element)
+                    // force rpgle to zoned if is number type
+                    val elementType = if (arrayType.element is NumberType) {
+                        NumberType(
```

Comment:
Use copy feature of dataclass:
`arrayType.element.copy(rpgType = RpgType.ZONED.rpgType)`

Reply:
Yes, done.